src/components/atoms/button/stories/button.stories.common.js

```diff
@@ -1,47 +1,132 @@
 import React from 'react';
 import { storiesOf } from '@storybook/react-native';
 import { action } from '@storybook/addon-actions';
-import { Text, View, Switch } from 'react-native';
-import { css } from '@emotion/native';
-import { useTheme } from 'emotion-theming';
-import { SafeView } from 'storybook/story-components-native';
-import { ThemeProvider, ThemeActionsContext } from 'store';
+import { withKnobs, text } from '@storybook/addon-knobs';
+import StoryPage, {
+  Description,
+  DocText,
+  DocSection,
+  DocItem,
+  DocCode,
+  DocExternalLink,
+  InlineCode,
+  PropTable,
+  StyleList,
+  TextList,
+  ThemeSwitcher
+} from 'storybook/story-components';
 
 import Button from '../';
 
-const wrapperStyle = css`
-  flex-direction: row;
-  justify-content: center;
-  align-items: center;
-  margin: 10px 0;
-`;
-
-const ThemeSwitcher = () => {
-  const { toggleTheme } = React.useContext(ThemeActionsContext);
-  const theme = useTheme();
-  return (
-    <View style={wrapperStyle}>
-      <Text>Switch theme</Text>
-      <Switch onValueChange={toggleTheme} value={theme.name !== 'light'} testID="theme-switch" />
-    </View>
-  );
-};
-
 storiesOf('Atoms/Button', module)
   .addDecorator(storyFn => (
-    <ThemeProvider>
-      <SafeView>{storyFn()}</SafeView>
-    </ThemeProvider>
+    <StoryPage title="Button component" url="components/atoms/button">
```

Comment:
You could likely abstract the structure you've come up with here further into their own components and then pass the `title`, `url` and `story` as props with the latter still being a child perhaps?

Reply:
I didn't create/edit the existing structure.  This came from `stories.web.js`. I just adapted it so that it would also work on native.

I agree with you though. This could be less verbose.

How about 

```
<StoryPage title="Button component" url="components/atoms/button" storyFn={storyFn}>
  An example story for a button
</StoryPage>
```